test/functional/e2e/user-management/user-profile-page-object.ts

```diff
@@ -0,0 +1,172 @@
+import puppeteer from 'puppeteer';
+
+const userProfileButton = '.col span';
+const userProfileOption = '.popover .MuiBox-root:nth-child(2) button';
+const userProfilePageName = 'h2 span';
+const editProfileButton = '.d-flex .align-items-end svg';
+const userProfileFormTitle = 'form h2 span';
+const fullNameField = 'input[name="displayName"]';
+const firstNameField = 'input[name="firstName"]';
+const lastNameField = 'input[name="lastName"]';
+const cityField = 'input[name="city"]';
+const phoneField = 'input[name="phone"]';
+const bioField = 'textarea[name="bio"]';
+const skillsField = 'input[placeholder="Communication, Blockchain"]';
+const keywordsField = 'input[placeholder="Innovation, AI, Technology"]';
+const genderMenu = 'select[name="gender"]';
+const selectGenderOptionMale = 'select[name="gender"] option:nth-child(2)';
+const countryDropdown = '.dropdown button';
+const countryDropdownMenuSearch = '.dropdown-menu input';
+const countryDropdownMenuFirstOption = '.dropdown-menu a';
+const saveButtonUpdateProfilePage = 'button[type="submit"]';
+const closeButtonUpdateProfilePage = `.d-flex.mt-4 button[type="button"]`;
+const successMessage = '.toast-header .text-success';
+const closeSuccessMessage = '.toast-header .close span:nth-child(1)';
+const addReferenceButton = '.flex-row-reverse button';
+const referenceName = 'input[name="references.0.name"]';
+const referenceValue = 'input[name="references.0.uri"]';
+const removeReferenceButton = '.align-items-end button';
+const userProilePageEntities = '.ct-card-body div span';
+
+export default class UserProfilePage {
+  page: puppeteer.Page | undefined;
+  value: string | undefined;
+
+  async verifyUserProfileTitle(page: puppeteer.Page, username: string) {
+    const usernameHeader = await page.$eval('h2 span', el =>
+      el.textContent?.trim()
+    );
+    if (usernameHeader !== username) {
+      throw console.error(
```

Comment:
throwing something other than `Error` is not the way to go. Throw an Error, log it, both , return nullable value or even better just return a boolean value, since you are verifyng something. I suggest the same for all your `verify `functions.

Reply:
`throw new Error('error message')`